crates/cext/src/param.rs

```diff
@@ -0,0 +1,1120 @@
+///
+///     QkParam *a = qk_param_new("a");
+///     QkParam *out = qk_param_new("");
+///     qk_param_acos(out, a);
+///
+/// # Safety
+///
+/// src should be valid pointer to parameter
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_param_acos(out: *mut Param, src: *const Param) -> ExitCode {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let out = unsafe { mut_ptr_as_ref(out) };
+    let src = unsafe { const_ptr_as_ref(src) };
+    match src {
+        Param::ParameterExpression(expr) => {
+            *out = Param::ParameterExpression(Arc::new(expr.acos()));
+            ExitCode::Success
+        }
+        Param::Float(f) => {
+            *out = Param::Float(f.acos());
+            ExitCode::Success
+        }
+        _ => ExitCode::ArithmeticError,
+    }
+}
+
+/// @ingroup QkParam
+/// calculate arctangent of a Param.
+///
+/// @param out A pointer to the QkParam to store a result.
+/// @param src A pointer to the QkParam.
+///
+/// @return an exit code
+///
+/// # Example
+///
+///     QkParam *a = qk_param_new("a");
+///     QkParam *out = qk_param_new("");
+///     qk_param_atan(out, a);
+///
+/// # Safety
+///
+/// src should be valid pointer to parameter
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_param_atan(out: *mut Param, src: *const Param) -> ExitCode {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let out = unsafe { mut_ptr_as_ref(out) };
+    let src = unsafe { const_ptr_as_ref(src) };
+    match src {
+        Param::ParameterExpression(expr) => {
+            *out = Param::ParameterExpression(Arc::new(expr.atan()));
+            ExitCode::Success
+        }
+        Param::Float(f) => {
+            *out = Param::Float(f.atan());
+            ExitCode::Success
+        }
+        _ => ExitCode::ArithmeticError,
+    }
+}
+
+/// @ingroup QkParam
+/// calculate log of a Param.
+///
+/// @param out A pointer to the QkParam to store a result.
+/// @param src A pointer to the QkParam.
+///
+/// @return an exit code
+///
+/// # Example
+///
+///     QkParam *a = qk_param_new("a");
+///     QkParam *out = qk_param_new("");
+///     qk_param_log(out, a);
+///
+/// # Safety
+///
+/// src should be valid pointer to parameter
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_param_log(out: *mut Param, src: *const Param) -> ExitCode {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let out = unsafe { mut_ptr_as_ref(out) };
+    let src = unsafe { const_ptr_as_ref(src) };
+    match src {
+        Param::ParameterExpression(expr) => {
+            *out = Param::ParameterExpression(Arc::new(expr.log()));
+            ExitCode::Success
+        }
+        Param::Float(f) => {
+            *out = Param::Float(f.ln());
+            ExitCode::Success
+        }
+        _ => ExitCode::ArithmeticError,
+    }
+}
+
+/// @ingroup QkParam
+/// calculate exponent of a Param.
```

Comment:
This sounds a bit unclear to me, how about
```suggestion
/// Apply the exponential function to a ``QkParam``.
```